### PR TITLE
introducing hookgen, an experimental tool for automatic hook(s) gener…

### PR DESCRIPTION
**Pull request by pdelewski**

### CODE REVIEW

File: src/HookGen/hookgen.stub.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/**
+ * @generate-class-entries
+ * @undocumentable
+ */
+
+function test1(): void {}
```
Comment (brettmc) [inline src/HookGen/hookgen.stub.php:8]: are these meant to be here?

File: src/HookGen/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+```
+phpize
+./configure
+make
+make install
+```
+
+## Usage
+
+php -dextension=hookgen.so [php app]
```
Comment (brettmc) [inline src/HookGen/README.md:17]: I've seen this named `hookgen` and `php_auto_instr_cfg`, maybe should have the same name everywhere for consistency?

File: src/HookGen/hookgen.stub.php (hunk 1 above)
Comment (pdelewski) [reply src/HookGen/hookgen.stub.php:8]: No, that's leftover, will remove it

File: src/HookGen/README.md (hunk 2 above)
Comment (pdelewski) [reply src/HookGen/README.md:17]: True, another leftover.

**Review by brettmc — APPROVED** [review]

**Pull request merged**